### DISCUSSION TITLE
Sonarr su Windows + Docker Desktop (parerei dopo 1 settimana)

### DISCUSSION THREAD

**Discussion by VegethB** [MainKronos/Sonarr-AnimeDownloader#11] [General]
Scrivo qui l'esito di una settimana (dopo il fix del problema #9 ) di test.
Il programma in se si comporta bene, ogni 30 minuti parte un check di tutte le serie in stato di monitoraggio con 1 episodio assente (minimo 1 ep). Se trova queste condizioni, parte il check sul sito di AnimeWorld per l'episodio mancante con relativo download e spostamento nella sua cartella.

Per quanto riguarda la tabella di gestione delle serie (JSON), L'interfaccia Web è molto semplice ma anche comoda. Graficamente si potrebbe espandere con le locandine delle serie ("grabbate" dal animeworld stesso o anilist / mal ecc.). Sarebbe più comodo poter vedere subito quali stagioni sono state aggiunte (solo le stagioni, niente link / url di AW). In una ipotetica Web UI con le locandine, si potrebbe mettere la locandina all'interno di un riquadro e quel riquadro, sotto, porterà le informazioni come: titolo e numero stagione. Questo poi se clickato allungherà la box inserendovi le informazioni aggiuntive come URL per AnimeWorld.
Sarebbe comodo inserire un metodo da WebUI per mettere anche quelle serie divise in due parti per AW ma viste come stagione singola per TheTVDB (evitando così di dover mettere mano al JSON).

Dal punto di vista dei LOG (su docker desktop), vengono ancora mostrate le serie monitorate e con almeno 1 episodio assente NON presenti nel JSON. Ragionandoci un po'... mi sono accorto che effettivamente può tornare utile sapere che queste serie (giustamente) sono state saltate perché non presenti nel JSON... Il problema è che se sono molte, il log diventa una giungla di testo rosso difficile da "scrollare". Quindi pensavo; se era possibile creare un LOG dinamico; In un file di configurazione da salvare nella cartella DATA in mount, Inserendo un parametro "TRUE / FALSE" per far si che nel LOG non vengano messi in output questi "errori". Essendo un parametro che solo un utente esperto può andare ad attivare, mi sembra l'opzione più giusta.

~~Infine, una feature assente che invece sarebbe opportuno inserire al più presto, è una regola di ridenominazione dei file scaricati dal programma. Prima cosa... non a tutti piace "s01e1" perché in certi casi su serie da 10+ ep il numero 1 (avvolte) viene messo dopo il 10 invece che prima (bulk rename, ecc). In più potrebbe essere necessario per l'utente finale inserire altri parametri nella regola, Esempio: "Nome Serie - s01e05 (in questo caso è minuscolo e non maiuscolo [S,s; E,e;]) - Nome Gruppo (in questo caso, AnimeWorld) - Titolo Episodio (magari preso da Sonarr che a sua volta viene preso da TheTVDB)" = "RWBY - s08e02 - AnimeWorld - Refuge.mp4".~~
Risolto nella 1.2.1 👍 (ma bello buggato... ha rinominato TUTTA la cartella RWBY invece che il singolo episodio scaricato, per fortuna mi sono accorto della cosa e ho fermato subito lo script 😨🥶)

In definitiva, il programma soddisfa per un 85% le aspettative generali.
Ci sono ancora un po' di cose da migliorare, WebUI, e LOG output ma già così siamo a buon punto 👍. 

Sono arrivato a trovare questo utilissimo programma nella disperata ricerca di un sostituto a YT-DL per Sonarr (che è fermo causa sviluppatore poco disponibile per problemi privati) e per alcune serie rilasciate solo su telegram.
**Consigliato a chi non pretende troppo e si accontenta dei formati HardSubs MP4 di AnimeWorld. Io l'ho uso per le serie su Telegram e tutte le serie VVVVID in simulcast.**

**Comment by MainKronos** [comment]
Grizie mille, sono felice che ti sia piaciuto👍.
Il problema della ridenominazione dei file scaricati è stato risolto nella versione `1.2.1`, adesso il nome è conforme alle impostazioni configurate in Sonarr.
Grazie per i consigli, provvederò ad implementarli👍

**Reply by VegethB** [reply]
Visto poco fa che è uscita la 1.2.1, Aggiorno subito 👍
